### PR TITLE
Diff 2.0

### PR DESCRIPTION
**Pull request by ZipFast**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: sync_diff_inspector/diff.go [hunk 1]
```diff
@@ -156,15 +161,53 @@ func (df *Diff) consume(chunk *chunk.Range) {
...
+				// TODO need table
+				Table: "",
+				// TODO translate Bound to string
+				UpperBound: "",
+				ChunkState: state,
+			},
+			// TODO need random value
+			RandomValue: make([][]string, 0),
+		}
+		randomNode.ID = chunk.ID
```
Comment (3pointer) [inline sync_diff_inspector/diff.go]: randomNode should be node

File: sync_diff_inspector/source/tidb.go [hunk 2]
```diff
@@ -93,17 +94,39 @@ func (s *TiDBChunksIterator) useBucket(diff *common.TableDiff) bool {
...
 	chunkSize := 1000
-	if s.useBucket(tableDiff) {
+	bucket := false
+	var node checkpoints.Node
+	if tableDiff.UseCheckpoint {
+		// TODO error handling
+		var err error
+		node, err = checkpoints.LoadChunks()
+		if err != nil {
+			tableDiff.UseCheckpoint = false
```
Comment (3pointer) [inline sync_diff_inspector/source/tidb.go:107]: should output warn log

File: sync_diff_inspector/source/tidb.go [hunk 3]
```diff
@@ -93,17 +94,39 @@ func (s *TiDBChunksIterator) useBucket(diff *common.TableDiff) bool {
...
+		} else {
+			switch node.(type) {
+			case *checkpoints.BucketNode:
+				bucket = true
+			case *checkpoints.RandomNode:
+				bucket = false
+			}
+		}
+	}
+	if (!tableDiff.UseCheckpoint && s.useBucket(tableDiff)) || bucket {
```
Comment (3pointer) [inline sync_diff_inspector/source/tidb.go:118]: `bucket` can be a part of `s.useBucket`

File: sync_diff_inspector/chunk/chunk.go [hunk 4]
```diff
@@ -20,6 +20,7 @@ import (
 
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/checkpoints"
```
Comment (3pointer) [inline sync_diff_inspector/chunk/chunk.go]: will this cause cycle dependence?

**Pull request merged**